docs/user-guide/deployments/deploy-on-kubernetes/common-helm-chart-configurations.md

```diff
@@ -0,0 +1,301 @@
+
+### Resource Configuration
+
+The top-level variable `base.podTemplate.main.resources` is used to globally configure resources for each role, as shown below:
+
+```yaml
+base:
+  podTemplate:
+    main:
+      resources:
+        requests:
+          memory: "1Gi"
+          cpu: "1"
+        limits:
+          memory: "2Gi"
+          cpu: "2"
+```
+
+If you want to configure different resources for each role in the cluster, you can use the `${role}.podTemplate.main.resources` field (where `role` can be `meta`, `frontend`, `datanode`, etc.). This field will **override** the top-level `base.podTemplate.main.resources` configuration, as shown below:
+
+```yaml
+base:
+  podTemplate:
+    main:
+      resources:
+        requests:
+          memory: "1Gi"
+          cpu: "1"
+        limits:
+          memory: "2Gi"
+          cpu: "2"
+
+frontend:
+  podTemplate:
+    main:
+      resources:
+        requests:
+          cpu: "2"
+          memory: "4Gi"
+        limits:
+          cpu: "4"
+          memory: "8Gi"
+```
+
+### Role Replicas Configuration
+
+For different roles, the number of replicas can be configured through the `${role}.replicas` field, as shown below:
+
+```yaml
+frontend:
+  replicas: 3
+
+datanode:
+  replicas: 3
+```
+
+Users can achieve horizontal scaling by configuring the number of replicas.
+
+### Environment Variable Configuration
+
+We can configure global environment variables through the `base.podTemplate.main.env` field, and configure different environment variables for each Role through the `${role}.podTemplate.main.env` field, as shown below:
+
+```yaml
+base:
+  podTemplate:
+    main:
+      env:
+        - name: GLOBAL_ENV
+          value: "global_value"
+
+frontend:
+  podTemplate:
+    main:
+      env:
+        - name: FRONTEND_ENV
+          value: "frontend_value"
+```
+
+### Injecting Configuration Files
+
+For different Role services, we can inject custom TOML configuration files through the `${role}.configData` field, as shown below:
+
+```yaml
+frontend:
+  configData: |
+    [[region_engine]]
+    [region_engine.mito]
+    # Number of region workers
+    num_workers = 8
+```
+
+Users can learn about GreptimeDB configuration options through [config.md](https://github.com/GreptimeTeam/greptimedb/blob/main/config/config.md).
+
+In addition to using the `${role}.configData` field to inject configuration files, users can also specify corresponding files through `${role}.configFile`, as shown below:
+
+```yaml
+frontend:
+  configFile: "configs/frontend.toml"
+```
+
+In this case, ensure that the configuration file path matches the directory where the `helm upgrade` command is executed.
+:::note
+User-injected configuration files have lower priority by default than configuration items managed by GreptimeDB Operator. Some configuration items can only be configured through GreptimeDB Operator, and these items are exposed by default in `values.yaml`.
```

Comment:
It would be better to clearly write out which items in "values.yaml" changing will have no effect.